### ISSUE TITLE
undefined symbol error when running Catch2 cmake module for discovering tests

### ISSUE THREAD

**Issue by tylerjw** [PickNikRobotics/pick_ik#62]
```
--- stderr: pick_ik
/home/tyler/code/ws_moveit2/build/pick_ik/tests/test-pick_ik: symbol lookup error: /home/tyler/code/ws_moveit2/build/pick_ik/tests/test-pick_ik: undefined symbol: _ZN7pick_ik17make_pose_cost_fnEN5Eigen9TransformIdLi3ELi1ELi0EEEmdd
CMake Error at /home/tyler/code/ws_moveit2/build/pick_ik/_deps/catch2-src/extras/CatchAddTests.cmake:60 (message):
  Error running test executable
  '/home/tyler/code/ws_moveit2/build/pick_ik/tests/test-pick_ik':

    Result: 127
    Output:
```

This is building on Humble the moveit2 tutorials with the sync from today.

Demangled that symbol is:
```c++
pick_ik::make_pose_cost_fn(Eigen::Transform<double, 3, 1, 0>, unsigned long, double, double)
```

**Comment by tylerjw** [comment]
I'm testing on ubuntu 22.04 with cmake 3.22.1. I'm invoking cmake through colcon.

I commented out the call to `catch_discover_tests` in the CMakeLists.txt in our tests directory and the project builds as expected. Then if I try to manually run the test executable I get this error as I would expect about not finding shared libraries:

```
./build/pick_ik/tests/test-pick_ik
./build/pick_ik/tests/test-pick_ik: error while loading shared libraries: libmoveit_exceptions.so.2.8.0: cannot open shared object file: No such file or directory
```

If I then source the workspace (`source install/setup.bash`) and re-invoke the test it runs as expected:

```
./build/pick_ik/tests/test-pick_ik
...
===============================================================================
All tests passed (124 assertions in 10 test cases)
```

**Comment by tylerjw** [comment]
This was caused by having `ros-humble-pick-ik` installed and the test linking with the library in `/opt/ros/humble` instead of the one built from source. I did a `apt remove` and the build error went away.

Closing the issue for now as this workaround is fine for now. Eventually it would be good to figure out why this is an issue and change our cmake so it links with the one built from source for the test.
